chore: upgrade to favicons v7

perrin4869 opened the pull request:
Does a few maintenance touches, but mainly adapts to the new `stream` API provided by `favicons v7`, which removed official support for `gulp`. The only thing necessary though was to replace the stream object with a `Vinyl` object instead.
This requires a major version bump

perrin4869 committed:
chore: upgrade to favicons v7

perrin4869 committed:
fix: tests are not passing on node 14

rejas commented:
Thanks for the PR. Would beside a major version bump be something necessary before releasing this (like readme update etc)?

perrin4869 committed:
update readme

perrin4869 committed:
final touches

perrin4869 commented:
Right! I updated the README to reflect the fact the module will now be ESM only and removed the example options that don't seem to be available on `favicons` anymore (I did a cursory check on the latest master). I think that should be it :)

rejas commented on .github/workflows/run-tests.yml:
```diff
@@ -15,7 +15,7 @@ jobs:
     strategy:
       matrix:
-        node-version: [12.x, 14.x, 16.x]
+        node-version: [16.x, 18.x]
```
just one more thing: why not have node 14 in here?

perrin4869 replied:
Oh, it failed to pass in my CI here: https://github.com/perrin4869/gulp-favicons/actions/runs/4281352674/jobs/7454270512
I didn't look into why really because node 14 becomes EOL next month anyways. If you prefer I can try to see what's breaking node 14 and add it back though

rejas replied:
I'm fine with it only being done on node16+
